proto/substrait/algebra.proto

```diff
@@ -364,6 +364,76 @@ message Rel {
   }
 }
 
+// A base table for writing. The list of string is used to represent namespacing (e.g., mydb.mytable).
+// This assumes shared catalog between systems exchanging a message.
+// it also includes a base schema, and default types
+message NamedTableWrite {
+  repeated string names = 1;
+  // The columns that will be modified (representing after-image of a schema change)
+  NamedStruct base_schema = 2;
+  // The default values for the columns (representing after-image of a schema change)
+  repeated Expression.Literal defaults = 3;
+
+  substrait.extensions.AdvancedExtension advanced_extension = 10;
+}
+
+message DdlRel {
+  // Definition of which type of scan operation is to be performed
+  oneof write_type {
+    NamedTableWrite named_table = 1;
+    ReadRel.ExtensionTable extension_table = 2;
+  }
+
+  //TODO add constraints/indexes/etc..?
+
+  // The type of operation to perform
+  DdlOp op = 3;
+
+  // The body of the CREATE VIEW / CTAS
+  Rel query = 4;
+
+  enum DdlOp {
+    DDL_OP_UNSPECIFIED = 0;
+    DDL_OP_CREATE_TABLE = 1;
+    DDL_OP_DROP_TABLE = 2;
+    DDL_OP_ALTER_TABLE = 3;
+
+    DDL_OP_CREATE_VIEW = 4;
+    DDL_OP_CREATE_OR_REPLACE = 5;
+    DDL_OP_DROP_VIEW = 6;
+  }
+}
+
+// The operator that modifies the schema/content of a database
+message WriteRel {
+  // Definition of which type of write operation is to be performed
+  oneof write_type {
+    NamedTableWrite named_table = 1;
+    ReadRel.LocalFiles local_files = 2;
```

Comment:
I don't think `ReadRel.LocalFiles` works here. For one thing, the `start` and `length` field are meaningless for writing, and likely so are future specific read options (e.g., for `ParquetReadOptions`. If the desire is to reuse fields, common fields (for reading and writing) should be refactored into a common place, such as `CommonRel.LocalFiles`.